Library/Homebrew/dependency_collector.rb

```diff
@@ -136,7 +136,7 @@ def parse_class_spec(spec, tags)
   end
 
   def ant_dep(spec, tags)
-    if MacOS.version >= :mavericks
+    if MacOS.version >= :mavericks || !OS.mac?
```

Comment:
I'm afraid when upstreaming changes you're going to need to do more stylistically involved changes than this. Instead of scattering `if OS.mac?` throughout the codebase the pattern is to make use of `Library/Homebrew/extend/os/linux` to override methods.